libraries/chain/apply_context.cpp

```diff
@@ -123,9 +123,22 @@ void apply_context::exec_one()
 
    action_trace& trace = trx_context.get_action_trace( action_ordinal );
    trace.receipt.emplace();
+
    action_receipt& r = *trace.receipt;
-   r.receiver         = receiver;
-   r.act_digest       = digest_type::hash(*act);
+   r.receiver        = receiver;
+
+   if( control.is_builtin_activated( builtin_protocol_feature_t::action_return_value ) ) {
+      trace.return_value = std::move(action_return_value);
+      r.act_digest = generate_action_digest(
+                        [this](const char* bytes, uint32_t len) {
+                           return trx_context.hash_with_checktime<digest_type>(bytes, len);
+                        },
+                        *act,
+                        trace.return_value
+                     );
+   } else {
+      r.act_digest = digest_type::hash(*act);
+   }
```

Comment:
Seems like this should be moved up into the try-catch.

Reply:
From Areg: "A light client following the blockchain will know when to switch the rules by detecting activation of the relevant protocol feature."

Reply:
Ok.